### PR TITLE
Generate keys in salt.utils.cloud.bootstrap if they're not already there

### PR DESCRIPTION
**Pull request by rallytime**
@techhat

### CODE REVIEW

File: salt/utils/cloud.py [hunk 1]
```diff
@@ -483,9 +510,6 @@ def bootstrap(vm_, opts):
         transport=opts.get('transport', 'zeromq')
     )
 
-    deployed = False
-    inline_script_deployed = False
```
Comment (techhat) [inline salt/utils/cloud.py]: Looks like you'll need to leave the `inline_script_deployed` line in, but removing the `deployed` line is okay.

File: salt/cloud/cli.py [hunk 2]
```diff
@@ -402,13 +402,6 @@ def run(self):
                     self.config['pki_dir'], vm_['pub_key'], key_id
                 )
 
-            if 'os' not in vm_:
```
Comment (techhat) [inline salt/cloud/cli.py]: I think you can take the above code block out here too, since you did move it into the other file.

Comment (rallytime) [inline salt/cloud/cli.py]: Oh, you're right. I was thinking we needed that block here for the drivers that don't call `salt.utils.cloud.bootstrap` but that should be covered in the `create` function in `salt/cloud/__init__.py`

File: salt/utils/cloud.py (hunk 1 above)
Comment (techhat) [inline salt/utils/cloud.py]: My bad. You can remove both.

**Pull request merged**